plugin.py

```diff
@@ -99,3 +115,74 @@ def find_package_dependency_dirs(self, py_ver: Tuple[int, int] = (3, 3)) -> List
         dep_dirs.insert(0, os.path.join(self.package_storage(), "resources", "typings", "sublime_text"))
 
         return [path for path in dep_dirs if os.path.isdir(path)]
+
+    @classmethod
+    def resolve_python_path_from_venv(
+        cls, settings: DottedDict, workspace_folders: List[WorkspaceFolder]
+    ) -> Optional[str]:
+        """
+        Resolves the python binary path depending on environment variables and files in the workspace.
+
+        See https://github.com/fannheyward/coc-pyright/blob/d58a468b1d7479a1b56906e386f44b997181e307/src/configSettings.ts#L47.  # noqa: E501
+        """
+
+        def binary_from_python_path(path: str) -> Optional[str]:
+            if sublime.platform() == "windows":
+                binary_path = os.path.join(path, "Scripts", "python.exe")
+            else:
+                binary_path = os.path.join(path, "bin", "python")
+
+            return binary_path if os.path.isfile(binary_path) else None
+
+        pythonPath = settings.get("python.pythonPath")
+        if pythonPath:
+            return pythonPath
+
+        virtual_env = os.getenv("VIRTUAL_ENV")
+        if virtual_env is not None:
+            return binary_from_python_path(virtual_env)
+
+        conda_prefix = os.getenv("CONDA_PREFIX")
+        if conda_prefix is not None:
+            return binary_from_python_path(conda_prefix)
+
+        if not len(workspace_folders):
+            return None
+        workspace_folder = workspace_folders[0].path
+
+        # Config file, venv resolution command, post-processing
+        venv_config_files = [
+            ("Pipfile", ["pipenv", "--py"], None),
+            ("poetry.lock", ["poetry", "env", "info", "-p"], binary_from_python_path),
+            (".python-version", ["pyenv", "which", "python"], None),
+        ]
+
+        for config_file, command, post_processing in venv_config_files:
+            full_config_file_path = os.path.join(workspace_folder, config_file)
+            if os.path.isfile(full_config_file_path):
+                try:
+                    pythonPath = subprocess.check_output(command, cwd=workspace_folder).decode("UTF-8").strip()
+                    return pythonPath if post_processing is None else post_processing(pythonPath)
+                except FileNotFoundError:
+                    print(
+                        "LSP: " + cls.name() + ": WARN: " + config_file + " detected but " + command[0] + " not found"
+                    )
+                except subprocess.CalledProcessError:
+                    print(
+                        "LSP: "
+                        + cls.name()
+                        + ": WARN: "
+                        + config_file
+                        + " detected but "
+                        + command
+                        + " exited with non-zero exit status"
+                    )
```

Comment:
Should be fine to skip `LSP: `prefix since the `cls.name()` says: `LSP-pyright`

Also suggested change to make it more compact:

```suggestion
                    print("{}: WARN: {} detected but {} not found".format(cls.name(), config_file, command[0]))
                except subprocess.CalledProcessError:
                    print("{}: WARN: {} detected but {} exited with non-zero exit status".format(
                        cls.name(), config_file, command))
```

Reply:
I found the console more consistent with the `LSP:` prefix.
For example:
```
LSP: LSP-pyright: supported code action kinds: ['quickfix', 'source.organizeImports']
LSP: LSP-pyright: registering capability: workspace.workspaceFolders
[...]
LSP: LSP-pyright: Using python path "/home/lucas/.cache/pypoetry/virtualenvs/predictor-6MLqJOL7-py3.10/bin/python"
```
But I am open to change it.

Reply:
Up to you. Personally I think it's redundant.
We have messages that start from `LSP:` but those specifically come from the LSP package so it makes sense in those cases.